js/derivadex.js

```diff
@@ -0,0 +1,599 @@
+                        'account/{trader}/strategy/{strategyId}/liquidations': 1,
+                        'account/{trader}/strategy/{strategyId}/order_book': 1,
+                        'account/{trader}/strategy/{strategyId}/order_intents': 1,
+                        'account/{trader}/strategy/{strategyId}/positions': 1,
+                        'account/{trader}/strategy/{strategyId}/strategy_updates': 1,
+                        'account/{trader}': 1,
+                        'account/{trader}/trader_updates': 1,
+                        'adl': 1,
+                        'ddx_fee_pool': 1,
+                        'epochs': 1,
+                        'fills': 1,
+                        'insurance_fund': 1,
+                        'liquidations': 1,
+                        'order_book': 1,
+                        'order_intents': 1,
+                        'positions': 1,
+                        'prices': 1,
+                        'specs': 1,
+                        'strategies': 1,
+                        'startegy_updates': 1,
+                        'trader_updates': 1,
+                        'traders': 1,
+                        'tx_logs': 1,
+                        'aggregations/collateral': 1,
+                        'aggregations/volume': 1,
+                        'aggregations/markets': 1,
+                        'markets/order_book/L2/{symbol}': 1,
+                    },
+                },
+                'private': {
+                    'get': {
+                        'apiKey': 5,
+                    },
+                    'post': {
+                        'apiKey': 5,
+                    },
+                },
+            },
+            // TODO: FILL OUT EXCEPTIONS
+            'exceptions': {
+                'exact': {
+                },
+                'broad': {
+                },
+            },
+            'precisionMode': TICK_SIZE,
+            'options': {
+                'defaultType': 'swap',
+                'defaultNetwork': 'ERC20',
+                'networks': {
+                    'ETH': 'ERC20',
+                },
+                'networksById': {
+                    'ETH': 'ERC20',
+                },
+            },
+            'fees': {
+                'trading': {
+                    'feeSide': 'get',
+                    'tierBased': false,
+                    'percentage': true,
+                    'taker': 0.002,
+                    'maker': 0.0,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchMarkets
+         * @description retrieves data on all markets for derivadex
+         * @param {object} params extra parameters specific to the derivadex api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        params['kind'] = 0;
+        const response = await this.publicGetSpecs (params);
```

Comment:
where is publicGetSpecs defined?

Reply:
the api methods are implicitly created by the api paths defined in the document